src/status_im/navigation/core.cljs

```diff
@@ -40,10 +40,8 @@
      (when (get views/screens view-id)
        ;;NOTE when back from the background on Android, this event happens for all screens, but we
        ;;need only for active one
-       (when (and @state/curr-modal (= @state/curr-modal view-id))
-         (effects/set-view-id view-id))
+       (effects/set-view-id view-id)
```

Comment:
Hi @briansztamfater, Thank you for fixing this issue.

This issue occurred because for `:navigate-to` and `open-modal` we are updating view-id in the events itself, but for `navigate-to-within-stack` we are not doing that.

I don't know why we originally didn't use this approach of updating view-id in `component-did-appear-listener`, but maybe there is a reason.

Should we do the same for `:navigate-to-within-stack` and just update the view-id in the navigation event? (safer)
(unless someone knows why we took this approach, and what issue the above comment/NOTE is referring to)

Reply:
On the surface, this look safe. So if you are confident, please feel free to use this one too.
(I am just little worried about NOTE part)

Reply:
Thank you @flexsurfer for adding context.

@briansztamfater Please can we use same approach also for `:navigate-to-within-stack` and update view-id in event itself?

Reply:
Hey @Parveshdhull @flexsurfer thanks for reviewing and adding context!

@Parveshdhull I am not sure if you mean to keep this change and also updating the view-id on `:navigate-to-within-stack`, or modify the approach only to update view-id within `:navigate-to-within-stack` event?

I tested the last option locally and it works when pushing screens, but on the other hand, it doesn't update properly when using `:navigate-back-within-stack` (we don't have the previous view-id to update db in the event), so I am not sure if we can do it without relying on `did-appear` listener. Let me know and I can make the proper changes :)

Reply:
Lets keep both, update in the event(for speed) and this change for back functionality.

Also, it seems  `(when (and @state/curr-modal (= @state/curr-modal view-id))` is outdated.
`@state/curr-modal` only holds values true/false when modal is open, not actual view-id. So its always false.

(Please also remove above comment/Note)

Reply:
Guys, please re-review.

I rebased and after merging #18778, which adds a prefix to some wallet screen ids, when navigating back `:view-id` wasn't updating for screens with a prefix, so needed to tweak the did-appear handler because the prefix is removed / ignored by `react-native-navigation`. Without this change, navigate back events won't update properly for screens with a prefix because the view-id returned from the did-appear listener comes without the prefix.

Hope this doesn't add much more complexity, in such case we may need to rollback #18778  :(

cc @Parveshdhull @flexsurfer @J-Son89

Reply:
Sure, if we can support both and look to update all keys then that would be great. It's just part of the process to qualifying screen keywords 👍